src/components/ApplicationDetails/WorkspaceSwitcher.tsx

```diff
@@ -1,35 +1,97 @@
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
 import { WorkspaceContext } from '../../utils/workspace-context-utils';
 import { ContextMenuItem, ContextSwitcher } from '../ContextSwitcher';
+import { ItemVisibility } from '../ContextSwitcher/context-switcher-utils';
+import {
+  ContextTab,
+  MenuTabs,
+  WORKSPACE_SWITCHER_STORAGE_KEY,
+} from '../ContextSwitcher/ContextSwitcher';
 
 export const WorkspaceSwitcher: React.FC<
   React.PropsWithChildren<{ selectedWorkspace?: string }>
 > = () => {
   const navigate = useNavigate();
   const { pathname } = useLocation();
-  const { workspace, workspaces } = React.useContext(WorkspaceContext);
+  const { workspace, konfluxWorkspaces = [] } = React.useContext(WorkspaceContext);
 
   const menuItems = React.useMemo(
-    () => workspaces?.map((app) => ({ key: app.metadata.name, name: app.metadata.name })) || [],
-    [workspaces],
+    () =>
+      (Array.isArray(konfluxWorkspaces) &&
+        konfluxWorkspaces.length > 0 &&
+        konfluxWorkspaces.map((ws) => ({
+          key: ws.status?.space.name ?? ws.metadata?.namespace,
+          name: ws.status?.space.name ?? ws.metadata?.namespace,
+          visibility: ws.spec?.visibility,
+        }))) ||
+      [],
+    [konfluxWorkspaces],
   );
-  const selectedItem = workspaces.find((item) => item.metadata.name === workspace) || workspaces[0];
+
+  const selectedItem =
+    konfluxWorkspaces?.find((item) => item.status?.space?.name === workspace) ||
+    (konfluxWorkspaces.length > 0 && konfluxWorkspaces[0]);
 
   const onSelect = (item: ContextMenuItem) => {
     // switch to new workspace but keep the first segment of the URL
-    navigate(
-      pathname.replace(/\/workspaces\/[-a-z0-9]+\/?([-a-z0-9]*).*/, `/workspaces/${item.name}/$1`),
-    );
+    if (pathname.indexOf('workspaces') === -1) {
+      navigate(`/application-pipeline/workspaces/${item.name}/applications`);
+    } else {
+      navigate(
+        pathname.replace(
+          /\/workspaces\/[-a-z0-9]+\/?([-a-z0-9]*).*/,
+          `/workspaces/${item.name}/$1`,
+        ),
+      );
+    }
   };
 
-  return workspaces.length > 0 ? (
+  const contextMenuItems = React.useMemo(
+    () => [
+      {
+        tabKey: ContextTab.Private,
+        tabName: MenuTabs.Private.name,
+        displayName: MenuTabs.Private.displayName,
+        menuItems: menuItems.filter((item) => item.visibility === ItemVisibility.PRIVATE),
+      },
+      {
+        tabKey: ContextTab.Public,
+        tabName: MenuTabs.Public.name,
+        displayName: MenuTabs.Public.displayName,
+        menuItems: menuItems.filter((item) => item.visibility === ItemVisibility.COMMUNITY),
+      },
+      {
+        tabKey: ContextTab.All,
+        tabName: MenuTabs.All.name,
+        displayName: MenuTabs.All.displayName,
+        menuItems,
+      },
+    ],
+    [menuItems],
+  );
+
+  return Array.isArray(konfluxWorkspaces) && konfluxWorkspaces.length > 0 ? (
```

Comment:
**[Suggestion, not necessary]**: You could try an early exit for this and extract some of the logic
For example:
```typescript
const { namespace } = selectedItem.metadata;

if (!Array.isArray(konfluxWorkspaces) || konfluxWorkspaces.length === 0) {
  return null;
}

const footer = (
  <Flex direction={{ default: 'column' }}>
    <FlexItem align={{ default: 'alignRight' }}>
      <Button variant="link" component={Link} to="/application-pipeline/workspace-list" isInline>
        View workspace list
      </Button>
    </FlexItem>
  </Flex>
);

return (
  <ContextSwitcher
    resourceType="workspace"
    menuItems={contextMenuItems}
    storageKey={WORKSPACE_SWITCHER_STORAGE_KEY}
    selectedItem={{ key: namespace, name: namespace }}
    onSelect={onSelect}
    footer={footer}
  />
);
```